LANDSCAPE/COMPARISONS/ARGOCD-FLUX/index.md

```diff
@@ -0,0 +1,275 @@
+
+With Argo CD you have more control over ordering of individual resources (i.e. specific YAML file) with [Syncwaves and Phases](https://argo-cd.readthedocs.io/en/stable/user-guide/sync-waves/). This allows you to apply each YAML manifest one at a time. Using Syncwaves and Phases, Argo CD will wait until one YAML is applied (and healthy) before moving onto applying the next one. 
+
+There is a controller that supplements Argo CD’s Application controller called [Application Sets](https://github.com/argoproj-labs/applicationset). This add-on lets users deploy many Applications across one or many managed clusters.
+
+
+# Secrets Management
+
+GitOps practice of storing the source of truth in git, has had some contention with respect to storing [Kubernetes Secrets](https://kubernetes.io/docs/concepts/configuration/secret/). This has to do with the fact that Secrets often contain sensitive 	information like passwords or tokens, and these secrets are only encoded.
+
+Both Flux and Argo CD have mechanisms in place to handle the encrypting of secrets.
+
+
+## FluxCD
+
+FluxCD provides a guide for managing encrypted secrets with [Mozilla SOPS](https://fluxcd.io/docs/guides/mozilla-sops/) or [Sealed Secrets](https://fluxcd.io/docs/guides/sealed-secrets/) so that Kubernetes secrets can be stored safely in Git, secured by any of OpenPGP, AWS KMS, GCP KMS and Azure Key Vault. Flux’s Kustomization objects natively support decrypting Mozilla SOPS secrets.
+
+
+## Argo CD
+
+Argo CD can also [integrate with a variety of Secret management tools](https://argo-cd.readthedocs.io/en/stable/operator-manual/secret-management/) to store these secrets in Git, safely. It leaves this choice up to the user. Some users have taken advantage of Argo CD’s plugin feature to create plugins, for example, the[ Argo CD Vault Plugin](https://github.com/IBM/argocd-vault-plugin). This “hands off” approach just treats secrets as another _Application_ in the cluster.
+
+
+# Webhook receivers
+
+Webhooks can be used to help with immediate reconciliation with traditional, event driven, workflows. This doesn’t take the place of the reconciliation loop, but rather acts as an “on demand” reconciliation if you don’t want to wait after making a commit to a repo.
+
+The two tools have similar approaches so we just highlight where to get more information.
+
+
+## FluxCD
+
+FluxCD includes support for incoming webhooks that make GitOps pull-based pipelines as fast and responsive as push-based pipelines. [Webhook receivers](https://fluxcd.io/docs/guides/webhook-receivers/) can notify the Flux controllers about changes in Git or Helm repositories, or other sources like Image Repositories.
+
+
+## Argo CD
+
+Argo CD also has a mechanism to receive [Webhooks](https://argo-cd.readthedocs.io/en/stable/operator-manual/webhook/) to shorten the time it takes to check for updates to a Git repository. Normally this is done every 3 minutes.
+
+
+# Alerting and notifications
```

Comment:
@christianh814 et al.

**In the context of:**

> (Proposed) GitOps Principle 5. **Operated in a closed loop**
>
>  Software agents observe desired state and meta-state to take actions based on policy when the desired state cannot be reached. This may include things like notifications, rollbacks, etc.

While this draft has [`# Alerting and notifications`](https://github.com/open-gitops/documents/blob/75fe9b46bbc484c58eec3441693185cf586c8717/LANDSCAPE/COMPARISONS/ARGOCD-FLUX/index.md#alerting-and-notifications)

❔ **Do Argo CD and Flux differ at all in the nuances of:**
• rollbacks
• when the desired state cannot be reached
• take actions based on policy
• etc.
?

**References:**
(Proposed) GitOps Principle 5. **Operated in a closed loop**
https://github.com/open-gitops/documents/pull/21#discussion_r709739072
https://github.com/open-gitops/documents/pull/22#discussion_r720456619
https://github.com/open-gitops/documents/issues/24#issuecomment-932754069

**Thank you.**

Reply:
@christianh814 et al.

**In the context of:**

> (Proposed) GitOps Principle 5. **Operated in a closed loop**
>
>  Software agents observe desired state and meta-state to take actions based on policy when the desired state cannot be reached. This may include things like notifications, rollbacks, etc.

While this draft has [`# Alerting and notifications`](https://github.com/open-gitops/documents/blob/75fe9b46bbc484c58eec3441693185cf586c8717/LANDSCAPE/COMPARISONS/ARGOCD-FLUX/index.md#alerting-and-notifications)

❔ **Do Argo CD and Flux differ at all in the nuances of:**
• rollbacks
• when the desired state cannot be reached
• take actions based on policy
• etc.
?

**References:**
(Proposed) GitOps Principle 5. **Operated in a closed loop**
https://github.com/open-gitops/documents/pull/21#discussion_r709739072
https://github.com/open-gitops/documents/pull/22#discussion_r720456619
https://github.com/open-gitops/documents/issues/24#issuecomment-932754069

**Thank you.**